addons/test_mail/models/test_mail_corner_case_models.py

```diff
@@ -86,6 +86,14 @@ class MailTestTrackMonetary(models.Model):
     company_currency = fields.Many2one("res.currency", string='Currency', related='company_id.currency_id', readonly=True, tracking=True)
     revenue = fields.Monetary('Revenue', currency_field='company_currency', tracking=True)
 
+class MailTestTrackMonetaryNoCompany(models.Model):
+    _name = 'mail.test.track.monetary.no_company'
+    _description = "Test Email Track Monetary No Company"
+    _inherit = ['mail.thread']
+
+    currency_id = fields.Many2one('res.currency', 'Currency', readonly=True, ondelete='set null',
```

Comment:
Actually is currency necessary at all ? Because what we want is to test the "no currency_field / no company" use case, unless I misunderstood something ?

Reply:
I don't think it's necessary, I'll leave it out

Reply:
i tried, but it throws an error. so i think still need currency

Reply:
I see as the behavior of this function, currency_field is indispensable when declaring fields Monetary.
https://github.com/odoo/odoo/blob/db52fb174461107e26b066147801f54a89d0dbf9/odoo/fields.py#L1478-L1481

Reply:
i think in this case we just need to remove company_id from getattr because currency_field always exists with field
@tde-banana-odoo Do you think it's reasonable?